## Practice Exercise 2

In this assignment, you will try to find some interesting insights into a few movies released between 1916 and 2016, using Python. You will have to download a movie dataset, write Python code to explore the data, gain insights into the movies, actors, directors, and collections, and submit the code.

#### Some tips before starting the assignment

1. Identify the task to be performed correctly, and only then proceed to write the required code. Don’t perform any incorrect analysis or look for information that isn’t required for the assignment.
2. In some cases, the variable names have already been assigned, and you just need to write code against them. In other cases, the names to be given are mentioned in the instructions. We strongly advise you to use the mentioned names only.
3. Always keep inspecting your data frame after you have performed a particular set of operations.
4. There are some checkpoints given in the IPython notebook provided. They're just useful pieces of information you can use to check if the result you have obtained after performing a particular task is correct or not.
5. Note that you will be asked to refer to documentation for solving some of the questions. That is done on purpose for you to learn new commands and also how to use the documentation.

In [1]:
# Import the numpy and pandas packages

import numpy as np
import pandas as pd

### Task 1: Reading and Inspection

**Subtask 1.1: Import and read**

Import and read the movie database. Store it in a variable called `movies`.

In [20]:
# Write your code for importing the csv file here
movies = pd.read_csv('Movies.csv')
movies

color      director_name  num_critic_for_reviews  duration  \
0     Color      James Cameron                   723.0     178.0   
1     Color     Gore Verbinski                   302.0     169.0   
2     Color         Sam Mendes                   602.0     148.0   
3     Color  Christopher Nolan                   813.0     164.0   
4     Color     Andrew Stanton                   462.0     132.0   
...     ...                ...                     ...       ...   
3848  Color      Shane Carruth                   143.0      77.0   
3849  Color   Neill Dela Llana                    35.0      80.0   
3850  Color   Robert Rodriguez                    56.0      81.0   
3851  Color       Edward Burns                    14.0      95.0   
3852  Color           Jon Gunn                    43.0      90.0   

      director_facebook_likes  actor_3_facebook_likes        actor_2_name  \
0                         0.0                   855.0    Joel David Moore   
1                       563.0                  1000.0       Orlando Bloom   
2                         0.0                   161.0        Rory Kinnear   
3                     22000.0                 23000.0      Christian Bale   
4                       475.0                   530.0     Samantha Morton   
...                       ...                     ...                 ...   
3848                    291.0                     8.0      David Sullivan   
3849                      0.0                     0.0     Edgar Tancangco   
3850                      0.0                     6.0     Peter Marquardt   
3851                      0.0                   133.0  Caitlin FitzGerald   
3852                     16.0                    16.0    Brian Herzlinger   

      actor_1_facebook_likes        gross  \
0                     1000.0  760505847.0   
1                    40000.0  309404152.0   
2                    11000.0  200074175.0   
3                    27000.0  448130642.0   
4                      640.0   73058679.0   
...                      ...          ...   
3848                   291.0     424760.0   
3849                     0.0      70071.0   
3850                   121.0    2040920.0   
3851                   296.0       4584.0   
3852                    86.0      85222.0   

                                   genres  ... num_user_for_reviews language  \
0         Action|Adventure|Fantasy|Sci-Fi  ...               3054.0  English   
1                Action|Adventure|Fantasy  ...               1238.0  English   
2               Action|Adventure|Thriller  ...                994.0  English   
3                         Action|Thriller  ...               2701.0  English   
4                 Action|Adventure|Sci-Fi  ...                738.0  English   
...                                   ...  ...                  ...      ...   
3848                Drama|Sci-Fi|Thriller  ...                371.0  English   
3849                             Thriller  ...                 35.0  English   
3850  Action|Crime|Drama|Romance|Thriller  ...                130.0  Spanish   
3851                         Comedy|Drama  ...                 14.0  English   
3852                          Documentary  ...                 84.0  English   

          country  content_rating       budget  title_year  \
0             USA           PG-13  237000000.0      2009.0   
1             USA           PG-13  300000000.0      2007.0   
2              UK           PG-13  245000000.0      2015.0   
3             USA           PG-13  250000000.0      2012.0   
4             USA           PG-13  263700000.0      2012.0   
...           ...             ...          ...         ...   
3848          USA           PG-13       7000.0      2004.0   
3849  Philippines       Not Rated       7000.0      2005.0   
3850          USA               R       7000.0      1992.0   
3851          USA       Not Rated       9000.0      2011.0   
3852          USA              PG       1100.0      2004.0   

     actor_2_face

**Subtask 1.2: Inspect the dataframe**

Inspect the dataframe's columns, shapes, variable types etc.

In [21]:
# Write your code for inspection here
movies.shape

(3853, 28)

#### <font color='red'>Question 1: How many rows and columns are present in the dataframe? </font>
-  <font color='red'>(3821, 26)</font>
-  <font color='red'>(3879, 28)</font>
-  <font color='red'>(3853, 28)</font>
-  <font color='red'>(3866, 26)</font>

In [22]:
null_counts = movies.isnull().sum()

# Identify columns with null values
columns_with_null = null_counts[null_counts > 0]

# Count the number of columns with null values
num_columns_with_null = len(columns_with_null)

# Sum up the counts of columns with null values
num_columns_with_null

12

#### <font color='red'>Question 2: How many columns have null values present in them? Try writing a code for this instead of counting them manually.</font>

-  <font color='red'>3</font>
-  <font color='red'>6</font>
-  <font color='red'>9</font>
-  <font color='red'>12</font>

### Task 2: Cleaning the Data

**Subtask 2.1: Drop unecessary columns**

For this assignment, you will mostly be analyzing the movies with respect to the ratings, gross collection, popularity of movies, etc. So many of the columns in this dataframe are not required. So it is advised to drop the following columns.
-  color
-  director_facebook_likes
-  actor_1_facebook_likes
-  actor_2_facebook_likes
-  actor_3_facebook_likes
-  actor_2_name
-  cast_total_facebook_likes
-  actor_3_name
-  duration
-  facenumber_in_poster
-  content_rating
-  country
-  movie_imdb_link
-  aspect_ratio
-  plot_keywords

In [23]:
# Check the 'drop' function in the Pandas library - dataframe.drop(list_of_unnecessary_columns, axis = )
# Write your code for dropping the columns here. It is advised to keep inspecting the dataframe after each set of operations
len(movies.columns.values)


28

#### <font color='red'>Question 3: What is the count of columns in the new dataframe? </font>
-  <font color='red'>10</font>
-  <font color='red'>13</font>
-  <font color='red'>15</font>
-  <font color='red'>17</font>

In [24]:
movies = movies.drop(columns=['color', 'director_facebook_likes', 'actor_1_facebook_likes', 'actor_2_facebook_likes', 'actor_3_facebook_likes', 'actor_2_name', 'cast_total_facebook_likes', 'actor_3_name', 'duration', 'facenumber_in_poster', 'content_rating', 'country', 'movie_imdb_link', 'aspect_ratio', 'plot_keywords'], axis=1)

In [25]:
movies.shape

(3853, 13)

**Subtask 2.2: Inspect Null values**

As you have seen above, there are null values in multiple columns of the dataframe 'movies'. Find out the percentage of null values in each column of the dataframe 'movies'. 

In [26]:
# Write you code here
movies.isnull().sum()


director_name             0
num_critic_for_reviews    1
gross                     0
genres                    0
actor_1_name              0
movie_title               0
num_voted_users           0
num_user_for_reviews      0
language                  3
budget                    0
title_year                0
imdb_score                0
movie_facebook_likes      0
dtype: int64

#### <font color='red'>Question 4: Which column has the highest percentage of null values? </font>
-  <font color='red'>language</font>
-  <font color='red'>genres</font>
-  <font color='red'>num_critic_for_reviews</font>
-  <font color='red'>imdb_score</font>

**Subtask 2.3: Fill NaN values**

You might notice that the `language` column has some NaN values. Here, on inspection, you will see that it is safe to replace all the missing values with `'English'`.

In [27]:
# Write your code for filling the NaN values in the 'language' column here
movies['language'] = movies['language'].fillna('English')
movies.head()


director_name  num_critic_for_reviews        gross  \
0      James Cameron                   723.0  760505847.0   
1     Gore Verbinski                   302.0  309404152.0   
2         Sam Mendes                   602.0  200074175.0   
3  Christopher Nolan                   813.0  448130642.0   
4     Andrew Stanton                   462.0   73058679.0   

                            genres     actor_1_name  \
0  Action|Adventure|Fantasy|Sci-Fi      CCH Pounder   
1         Action|Adventure|Fantasy      Johnny Depp   
2        Action|Adventure|Thriller  Christoph Waltz   
3                  Action|Thriller        Tom Hardy   
4          Action|Adventure|Sci-Fi     Daryl Sabara   

                                 movie_title  num_voted_users  \
0                                    Avatar            886204   
1  Pirates of the Caribbean: At World's End            471220   
2                                   Spectre            275868   
3                     The Dark Knight Rises           1144337   
4                               John Carter            212204   

   num_user_for_reviews language       budget  title_year  imdb_score  \
0                3054.0  English  237000000.0      2009.0         7.9   
1                1238.0  English  300000000.0      2007.0         7.1   
2                 994.0  English  245000000.0      2015.0         6.8   
3                2701.0  English  250000000.0      2012.0         8.5   
4                 738.0  English  263700000.0      2012.0         6.6   

   movie_facebook_likes  
0                 33000  
1                     0  
2                 85000  
3                164000  
4                 24000

In [29]:
movies[movies['language'] == 'English']['language'].value_counts()

English    3674
Name: language, dtype: int64

#### <font color='red'>Question 5: What is the count of movies made in English language after replacing the NaN values with English? </font>
-  <font color='red'>3670</font>
-  <font color='red'>3674</font>
-  <font color='red'>3668</font>
-  <font color='red'>3672</font>

### Task 3: Data Analysis

**Subtask 3.1: Change the unit of columns**

Convert the unit of the `budget` and `gross` columns from `$` to `million $`.

In [31]:
# Write your code for unit conversion here
movies['budget'] = movies['budget'].apply(lambda x: x/1000000)
movies['budget'].describe()

count     3853.000000
mean        45.248166
std        223.420850
min          0.000218
25%         10.000000
50%         24.000000
75%         50.000000
max      12215.500000
Name: budget, dtype: float64

In [32]:
movies['gross'] = movies['gross'].apply(lambda x: x/1000000)
movies['gross'].describe()

count    3853.000000
mean       50.965473
std        69.320329
min         0.000162
25%         6.830957
50%        27.900000
75%        65.500000
max       760.505847
Name: gross, dtype: float64

**Subtask 3.2: Find the movies with highest profit**

   1. Create a new column called `profit` which contains the difference of the two columns: `gross` and `budget`.
   2. Sort the dataframe using the `profit` column as reference. (Find which command can be used here to sort entries from the documentation)
   3. Extract the top ten profiting movies in descending order and store them in a new dataframe - `top10`

In [33]:
# Write your code for creating the profit column here
movies['profit']=movies['gross']-movies['budget']
movies['profit'].describe()

count     3853.000000
mean         5.717307
std        227.148466
min     -12213.298588
25%        -10.227031
50%          0.865931
75%         24.699513
max        523.505847
Name: profit, dtype: float64

In [37]:
# Write your code for sorting the dataframe here
moviesByProfit = movies.sort_values('profit', ascending=False)
moviesByProfit.head(10)

director_name  num_critic_for_reviews       gross  \
0         James Cameron                   723.0  760.505847   
28      Colin Trevorrow                   644.0  652.177271   
25        James Cameron                   315.0  658.672302   
2704       George Lucas                   282.0  460.935665   
2748   Steven Spielberg                   215.0  434.949459   
16          Joss Whedon                   703.0  623.279547   
482        Roger Allers                   186.0  422.783777   
230        George Lucas                   320.0  474.544677   
64    Christopher Nolan                   645.0  533.316061   
419           Gary Ross                   673.0  407.999255   

                                        genres         actor_1_name  \
0              Action|Adventure|Fantasy|Sci-Fi          CCH Pounder   
28            Action|Adventure|Sci-Fi|Thriller  Bryce Dallas Howard   
25                               Drama|Romance    Leonardo DiCaprio   
2704           Action|Adventure|Fantasy|Sci-Fi        Harrison Ford   
2748                             Family|Sci-Fi         Henry Thomas   
16                     Action|Adventure|Sci-Fi      Chris Hemsworth   
482   Adventure|Animation|Drama|Family|Musical    Matthew Broderick   
230            Action|Adventure|Fantasy|Sci-Fi      Natalie Portman   
64                 Action|Crime|Drama|Thriller       Christian Bale   
419            Adventure|Drama|Sci-Fi|Thriller    Jennifer Lawrence   

                                     movie_title  num_voted_users  \
0                                        Avatar            886204   
28                               Jurassic World            418214   
25                                      Titanic            793059   
2704         Star Wars: Episode IV - A New Hope            911097   
2748                 E.T. the Extra-Terrestrial            281842   
16                                 The Avengers            995415   
482                               The Lion King            644348   
230   Star Wars: Episode I - The Phantom Menace            534658   
64                              The Dark Knight           1676169   
419                            The Hunger Games            701607   

      num_user_for_reviews language  budget  title_year  imdb_score  \
0                   3054.0  English   237.0      2009.0         7.9   
28                  1290.0  English   150.0      2015.0         7.0   
25                  2528.0  English   200.0      1997.0         7.7   
2704                1470.0  English    11.0      1977.0         8.7   
2748                 515.0  English    10.5      1982.0         7.9   
16                  1722.0  English   220.0      2012.0         8.1   
482                  656.0  English    45.0      1994.0         8.5   
230                 3597.0  English   115.0      1999.0         6.5   
64                  4667.0  English   185.0      2008.0         9.0   
419                 1959.0  English    78.0      2012.0         7.3   

      movie_facebook_likes      profit  
0                    33000  523.505847  
28                  150000  502.177271  
25                   26000  458.672302  
2704                 33000  449.935665  
2748                 34000  424.449459  
16                  123000  403.279547  
482                  17000  377.783777  
230                  13000  359.544677  
64                   37000  348.316061  
419                 140000  329.999255

**Checkpoint:** You might spot two movies directed by `James Cameron` in the list.

#### <font color='red'>Question 6: Which movie is ranked 5th from the top in the list obtained? </font>
-  <font color='red'>E.T. the Extra-Terrestrial</font>
-  <font color='red'>The Avengers</font>
-  <font color='red'>The Dark Knight</font>
-  <font color='red'>Titanic</font>

**Subtask 3.3: Find IMDb Top 250**

Create a new dataframe `IMDb_Top_250` and store the top 250 movies with the highest IMDb Rating (corresponding to the column: `imdb_score`). Also make sure that for all of these movies, the `num_voted_users` is greater than 25,000. 

Also add a `Rank` column containing the values 1 to 250 indicating the ranks of the corresponding films.

In [41]:
# Write your code for extracting the top 250 movies as per the IMDb score here. Make sure that you store it in a new dataframe 
# and name that dataframe as 'IMDb_Top_250'
IMDb_Top_250 = movies[movies['num_voted_users']>25000]
IMDb_Top_250 = IMDb_Top_250.sort_values(by='imdb_score', ascending=False).head(250)
IMDb_Top_250

director_name  num_critic_for_reviews       gross  \
1795        Frank Darabont                   199.0   28.341469   
3016  Francis Ford Coppola                   208.0  134.821952   
64       Christopher Nolan                   645.0  533.316061   
2543  Francis Ford Coppola                   149.0   57.300000   
325          Peter Jackson                   328.0  377.019252   
...                    ...                     ...         ...   
67             Jon Favreau                   486.0  318.298180   
73              Doug Liman                   585.0  100.189501   
3052           Yash Chopra                    29.0    2.921738   
95           Peter Jackson                   645.0  303.001229   
3170  Christophe Barratier                   112.0    3.629758   

                              genres           actor_1_name  \
1795                     Crime|Drama         Morgan Freeman   
3016                     Crime|Drama              Al Pacino   
64       Action|Crime|Drama|Thriller         Christian Bale   
2543                     Crime|Drama         Robert De Niro   
325   Action|Adventure|Drama|Fantasy          Orlando Bloom   
...                              ...                    ...   
67           Action|Adventure|Sci-Fi      Robert Downey Jr.   
73           Action|Adventure|Sci-Fi             Tom Cruise   
3052           Drama|Musical|Romance         Shah Rukh Khan   
95                 Adventure|Fantasy           Aidan Turner   
3170                     Drama|Music  Jean-Baptiste Maunier   

                                         movie_title  num_voted_users  \
1795                       The Shawshank Redemption           1689764   
3016                                  The Godfather           1155770   
64                                  The Dark Knight           1676169   
2543                         The Godfather: Part II            790926   
325   The Lord of the Rings: The Return of the King           1215718   
...                                              ...              ...   
67                                         Iron Man            696338   
73                                 Edge of Tomorrow            431620   
3052                                     Veer-Zaara             34449   
95                The Hobbit: An Unexpected Journey            637246   
3170                                     The Chorus             44151   

      num_user_for_reviews language  budget  title_year  imdb_score  \
1795                4144.0  English    25.0      1994.0         9.3   
3016                2238.0  English     6.0      1972.0         9.2   
64                  4667.0  English   185.0      2008.0         9.0   
2543                 650.0  English    13.0      1974.0         9.0   
325                 3189.0  English    94.0      2003.0         8.9   
...                    ...      ...     ...         ...         ...   
67                  1055.0  English   140.0      2008.0         7.9   
73                   741.0  English   178.0      2014.0         7.9   
3052                 119.0    Hindi     7.0      2004.0         7.9   
95                  1367.0  English   180.0      2012.0         7.9   
3170                 110.0   French     5.5      2004.0         7.9   

      movie_facebook_likes      profit  
1795                108000    3.341469  
3016                 43000  128.821952  
64                   37000  348.316061  
2543                 14000   44.300000  
325                  16000  283.019252  
...                    ...         ...  
67                   10000  178.298180  
73                   77000  -77.810499  
3052                  2000   -4.078262  
95                  166000  123.001229  
3170                     0   -1.870242  

[250 rows x 14 columns]

#### <font color='red'>Question 7: Suppose movies are divided into 5 buckets based on the IMDb ratings: </font>
-  <font color='red'>7.5 to 8</font>
-  <font color='red'>8 to 8.5</font>
-  <font color='red'>8.5 to 9</font>
-  <font color='red'>9 to 9.5</font>
-  <font color='red'>9.5 to 10</font>

<font color = 'red'> Which bucket holds the maximum number of movies from *IMDb_Top_250*? </font>

In [44]:
bin_edges = [7.5, 8, 8.5, 9, 9.5, 10]

# Define labels for the bins
bin_labels = ['[7.5, 8]', '[8, 8.5]', '[8.5, 9]', '[9, 9.5]', '[9.5, 10]']

# Create a new column with the bin labels based on imdb_score
IMDb_Top_250['imdb_score_bucket'] = pd.cut(IMDb_Top_250['imdb_score'], bins=bin_edges, labels=bin_labels, right=False)
IMDb_Top_250['imdb_score_bucket'].value_counts()

[8, 8.5]     159
[7.5, 8]      45
[8.5, 9]      42
[9, 9.5]       4
[9.5, 10]      0
Name: imdb_score_bucket, dtype: int64

**Subtask 3.4: Find the critic-favorite and audience-favorite actors**

   1. Create three new dataframes namely, `Meryl_Streep`, `Leo_Caprio`, and `Brad_Pitt` which contain the movies in which the actors: 'Meryl Streep', 'Leonardo DiCaprio', and 'Brad Pitt' are the lead actors. Use only the `actor_1_name` column for extraction. Also, make sure that you use the names 'Meryl Streep', 'Leonardo DiCaprio', and 'Brad Pitt' for the said extraction.
   2. Append the rows of all these dataframes and store them in a new dataframe named `Combined`.
   3. Group the combined dataframe using the `actor_1_name` column.
   4. Find the mean of the `num_critic_for_reviews` and `num_user_for_review` and identify the actors which have the highest mean.

In [ ]:
# Write your code for creating three new dataframes here
Meryl_Streep = # Include all movies in which Meryl_Streep is the lead

In [ ]:
Leo_Caprio = # Include all movies in which Leo_Caprio is the lead

In [ ]:
Brad_Pitt = # Include all movies in which Brad_Pitt is the lead

In [46]:
# Write your code for combining the three dataframes here
combined = movies[movies['actor_1_name'].isin(['Meryl Streep','Leonardo DiCaprio','Brad Pitt'])]
combined

director_name  num_critic_for_reviews       gross  \
25            James Cameron                   315.0  658.672302   
49             Baz Luhrmann                   490.0  144.812796   
94        Christopher Nolan                   642.0  292.568851   
97            David Fincher                   362.0  127.490802   
142       Wolfgang Petersen                   220.0  133.228348   
173   Alejandro G. Iñárritu                   556.0  183.635922   
243       Steven Soderbergh                   198.0  125.531634   
244              Doug Liman                   233.0  186.336103   
246         Martin Scorsese                   267.0  102.608827   
283       Quentin Tarantino                   765.0  162.804648   
293            Edward Zwick                   166.0   57.366262   
294         Martin Scorsese                   606.0  116.866727   
312         Martin Scorsese                   233.0   77.679638   
347         Martin Scorsese                   352.0  132.373442   
367              Tony Scott                   142.0    0.026871   
383       Steven Soderbergh                   186.0  183.405771   
392            Nancy Meyers                   187.0  112.703470   
431         Martin Scorsese                   490.0  127.968405   
448              David Ayer                   406.0   85.707116   
579     Jean-Jacques Annaud                    76.0   37.901509   
608            Ridley Scott                   238.0   39.380442   
646           David Fincher                   315.0   37.023395   
749         Patrick Gilmore                    98.0   26.288320   
859        Steven Spielberg                   194.0  164.435221   
887             Neil Jordan                   120.0  105.264608   
934             Danny Boyle                   118.0   39.778599   
1038          Curtis Hanson                    42.0   46.815748   
1046             Sam Mendes                   323.0   22.877808   
1132            Nora Ephron                   252.0   94.125426   
1322          David Frankel                   208.0  124.732962   
1336        Randall Wallace                    83.0   56.876365   
1364         Clint Eastwood                   392.0   37.304950   
1390         Robert Redford                   227.0   14.998070   
1397        Terrence Malick                   584.0   13.303319   
1457              Sam Raimi                    63.0   18.636537   
1471         Sydney Pollack                    66.0   87.100000   
1514          David Frankel                   234.0   63.536011   
1563          Carl Franklin                    64.0   23.209440   
1608         Andrew Dominik                   273.0    3.904982   
1784         Stephen Daldry                   174.0   41.597830   
1909             Jerry Zaks                    45.0   12.782508   
2030  Alejandro G. Iñárritu                   285.0   34.300771   
2146    Angelina Jolie Pitt                   131.0    0.531009   
2428         Andrew Dominik                   414.0   14.938570   
2484           Baz Luhrmann                   106.0   46.338728   
2500         Phyllida Lloyd                   331.0   29.959436   
2602             Tony Scott                   122.0   12.281500   
2793          Robert Altman                   211.0   20.338609   
3025           Baz Luhrmann                   490.0  144.812796   

                                                 genres       actor_1_name  \
25                                        Drama|Romance  Leonardo DiCaprio   
49                                        Drama|Romance  Leonardo DiCaprio   
94                     Action|Adventure|Sci-Fi|Thriller  Leonardo DiCaprio   
97                                Drama|Fantasy|Romance          Brad Pitt   
142                                           Adventure          Brad Pitt   
173                    Adventure|Drama|Thriller|Western  Leonardo DiCaprio   
243                                      Crime|Thriller          Brad Pitt   
244                Action|Comedy|Crime|

In [47]:
# Write your code for grouping the combined dataframe here
combined_mean = combined.groupby(by='actor_1_name').mean()
combined_mean

/var/folders/3p/bp6sw9nn0_s1z8g3v89b2vq40000gp/T/ipykernel_7452/1016322582.py:2: FutureWarning: The default value of numeric_only in DataFrameGroupBy.mean is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  combined_mean = combined.groupby(by='actor_1_name').mean()


num_critic_for_reviews       gross  num_voted_users  \
actor_1_name                                                             
Brad Pitt                      245.000000   66.321449    283583.823529   
Leonardo DiCaprio              330.190476  125.741989    441958.857143   
Meryl Streep                   181.454545   59.919727     73545.545455   

                   num_user_for_reviews     budget   title_year  imdb_score  \
actor_1_name                                                                  
Brad Pitt                    742.352941  69.294118  2004.352941    7.152941   
Leonardo DiCaprio            914.476190  82.452381  2005.476190    7.495238   
Meryl Streep                 297.181818  34.454545  2003.545455    6.745455   

                   movie_facebook_likes     profit  
actor_1_name                                        
Brad Pitt                  14051.764706  -2.972669  
Leonardo DiCaprio          52190.476190  43.289608  
Meryl Streep                2934.090909  25.465182

In [ ]:
# Write the code for finding the mean of critic reviews and audience reviews here


#### <font color='red'>Question 8: Which actor is highest rated among the three actors according to the user reviews? </font>
-  <font color='red'>Meryl Streep</font>
-  <font color='red'>Leonardo DiCaprio</font>
-  <font color='red'>Brad Pitt</font>

#### <font color='red'>Question 9: Which actor is highest rated among the three actors according to the critics?</font>
-  <font color='red'>Meryl Streep</font>
-  <font color='red'>Leonardo DiCaprio</font>
-  <font color='red'>Brad Pitt</font>